src/language/walker/walker.ts

```diff
@@ -17,10 +17,59 @@
 
 import * as ts from "typescript";
 
-import {RuleFailure} from "../rule/rule";
+import {Fix, IOptions, Replacement, RuleFailure} from "../rule/rule";
+import {IWalker} from "./walker";
 
 export interface IWalker {
     getSourceFile(): ts.SourceFile;
-    walk(node: ts.Node): void;
+    walk(sourceFile: ts.SourceFile): void;
     getFailures(): RuleFailure[];
 }
+
+export abstract class AbstractWalker implements IWalker {
+    public readonly options: ReadonlyArray<any>;
+    public readonly ruleName: string;
+    private limit: number;
+    private failures: RuleFailure[];
+
+    constructor(public readonly sourceFile: ts.SourceFile, options: IOptions) {
+        this.failures = [];
+        this.options = options.ruleArguments;
+        this.limit = sourceFile.getFullWidth();
+        this.ruleName = options.ruleName;
+    }
+
+    public abstract walk(sourceFile: ts.SourceFile): void;
+
+    public getSourceFile() {
+        return this.sourceFile;
+    }
+
+    public getFailures() {
+        return this.failures;
+    }
+
+    public hasOption(option: any) {
```

Comment:
I like that idea. I have the following in mind:
```typescript
class AbstractWalker<T> {
    ...
    constructor(public readonly sourceFile: ts.SourceFile, public readonly options: T) {
        ...
    }
}
```
This way you have the choice to parse the options in `Rule` or inside the constructor of your specific walker implementation ... or you can opt out completely by setting the type parameter to `any`;